influxql/query_executor.go

```diff
@@ -349,44 +349,50 @@ func (e *QueryExecutor) query(qid uint64) (*QueryTask, bool) {
 //
 // After a query finishes running, the system is free to reuse a query id.
 func (e *QueryExecutor) attachQuery(q *Query, database string, interrupt <-chan struct{}) (uint64, *QueryTask, error) {
-	e.mu.Lock()
-	defer e.mu.Unlock()
+	qid, task, err := func() (uint64, *QueryTask, error) {
```

Comment:
I'm fine with changing it from an anonymous function, but manually unlocking the mutex seems risky to me. The function is of significant enough complexity and also contains allocations that it would be risky if a panic were to occur because of something like out of memory or programmer error.